### ISSUE TITLE
Add support for Camversity

### ISSUE THREAD

**Issue by sxnfxsed** [0xboobface/ctbrec#140]
Camversity is a streaming site that offers free shows. There are a couple big names on there that I'd like to record automatically. It looks like they use some websocket shit, dunno though just had a look. Happy new years!

**Comment by 0xboobface** [comment]
Had a quick look. You are right with the websocket streaming, but you can trick them into HLS streaming with the right http user agent. Looks doable.

**Comment by sxnfxsed** [comment]
Oh damn youre right, nice catch. That'll hold me off until it can be integrated into ctbrec lmao

**Comment by hooligan1** [comment]
hi, I was also looking into camv but I got stuck on the weird websocket streaming. sometimes it forces you into that mode so I was looking for a way to always get the HLS stream. could you elaborate on the the user agent thing?

**Comment by 0xboobface** [comment]
I get an HLS stream, if I set the HTTP header "User-Agent" to "Mozilla/5.0 (iPhone; CPU OS 10_14 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/11.1.1 Mobile/14E304 Safari/605.1.15"
You can use browser extensions to change your user-agent.

**Comment by hooligan1** [comment]
thanks. unfortunately there's no way to get the HLS version on paid streams. I wish I could somehow find a way to record the websocket stream but I haven't so far

**Comment by 0xboobface** [comment]
LiveJasmin sents the video as mp4 chunks encoded as binary websocket frames. Those can easily be saved in one file and you end up with one mp4 recording. I haven't had a look at camversity, but I assume they use something similar.

**Comment by hooligan1** [comment]
yes, from what I can tell it works similarly. if you can adapt the jasmin one to work on camv that would be amazing, even the regular HLS version would be great for automated recording

**Comment by megaraus** [comment]
I have seen 1 site that auto records and posts camversity shows, but no idea how they're doing it

**Comment by megaraus** [comment]
Also interested in camversity, grabbing HLS stream does work though.

**Comment by Samoanian** [comment]
Yes, grabbing HLS stream is OK, but you must do it manually. Would be great monitoring ur fav models 24/7, so that it automatically records when they're online.
